src/frame/worker/GroupCrypto.ts

```diff
@@ -28,23 +28,33 @@ export function createGroup(userPublicKey: PublicKey<Uint8Array>, signingKeys: S
 }
 
 /**
- * Generate encrypted access keys to the groups private key for the provided list of users in order to make them admins of the group.
+ * Generate schnorr siganture and encrypted access keys to the groups private key for the provided list of users in order to make them admins of the group.
  * @param {TransformedEncryptedMessage} encryptedGroupPrivateKey Groups encrypted private key
+ * @param {PublicKey<string>}           groupPublicKey           Group public key of the group being added to
+ * @param {string}                      groupID                  Group Id of the group being added to
  * @param {UserOrGroupPublicKey[]}      userKeyList              List of user public keys to encrypt group private key to
  * @param {PrivateKey<Uint8Array>}      adminPrivateKey          Private key for the admin who is adding the other admins. The current user performing this operation.
  * @param {SigningKeyPair}              signingKeys              Signing keys for the admin adding the other admins. The current user performing this operation.
  */
 export function addAdminsToGroup(
     encryptedGroupPrivateKey: TransformedEncryptedMessage,
+    groupPublicKey: PublicKey<string>,
+    groupID: string,
     userKeyList: UserOrGroupPublicKey[],
     adminPrivateKey: PrivateKey<Uint8Array>,
     signingKeys: SigningKeyPair
 ) {
     return loadRecrypt()
         .flatMap((Recrypt) => {
-            return Recrypt.decryptPlaintext(encryptedGroupPrivateKey, adminPrivateKey).flatMap(([plaintext]) =>
-                Recrypt.encryptPlaintextToList(plaintext, userKeyList, signingKeys)
-            );
+            return Recrypt.decryptPlaintext(encryptedGroupPrivateKey, adminPrivateKey).flatMap(([plaintext, key]) => {
+                return Future.gather2(
+                    Recrypt.encryptPlaintextToList(plaintext, userKeyList, signingKeys),
+                    Future.of(Recrypt.schnorrSignUtf8String(key, groupPublicKey, groupID))
```

Comment:
I'm a bit confused why you did this in a Future.of? Can you just remove the future wrapping and put the signature calculation in the map call below?

Reply:
Ya, sorry for the poor guidance on this, I wasn't thinking it through. Do what Colt said.

Reply:
done